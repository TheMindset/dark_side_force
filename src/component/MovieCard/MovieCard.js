import React from 'react'
import './MovieCard.scss'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const MovieCard = ({ movie, reachMovieCharacters }) => {
  const { title, episode, releaseYear, id } = movie
  return (
    <div className="movie-card">
      <h2> {title} </h2>
      <h3> Episode: {episode} </h3>
      <h3> Released in {releaseYear} </h3>
      <Link onClick={reachMovieCharacters} to={`/movies/${id}`}>
        <button id={id}>Explore Characters</button>
      </Link>
    </div>
  )
}

export default MovieCard

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  reachMovieCharacters: PropTypes.func.isRequired
}