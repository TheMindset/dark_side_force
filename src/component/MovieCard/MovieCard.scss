* {
  box-sizing: border-box;
}

.movie-card {
  background-color: #04243d;
  border-radius: 20px;
  border: 8px, ridge, white;
  color: white;
  display: flex;
  flex-direction: column;
  font-size: 21px;
  justify-content: space-between;
  margin: 15px;
  padding: 1.5rem;
  text-align: center;
}

button {
  background: white;
  border-radius: 3px;
  color: #04243d;
  border: 2px solid white;
  display: inline-block;
  font-size: 1.2rem;
  font-weight: bold;
  margin-top: 15px;
  padding: 0.4rem 1.3rem;
  transition: opacity 0.2s ease-in;
  &:hover {
    cursor: pointer;
  }
}

