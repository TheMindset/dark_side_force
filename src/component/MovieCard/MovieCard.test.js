import React from 'react'
import { shallow } from 'enzyme'
import MovieCard from './MovieCard'


describe('MovieCard', () => {
  let wrapper, mockMovie, mockReachMovieCharacter

  beforeEach(() => {
    mockReachMovieCharacter = jest.fn()
    mockMovie = {
      title: 'The GodFather',
      episode: 1,
      releaseYear: 1972,
      id:  2      
    }
    wrapper = shallow(<MovieCard movie={mockMovie} reachMovieCharacters={mockReachMovieCharacter} />)
  })

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should call reachMovieCharacters on Click event', () => {
    wrapper.find('Link').simulate('click')

    expect(mockReachMovieCharacter).toHaveBeenCalled()
  })
})
