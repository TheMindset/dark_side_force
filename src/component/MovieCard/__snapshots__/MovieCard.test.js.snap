// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<div
  className="movie-card"
>
  <h2>
     
    The GodFather
     
  </h2>
  <h3>
     Episode: 
    1
     
  </h3>
  <h3>
     Released in 
    1972
     
  </h3>
  <Link
    onClick={[MockFunction]}
    to="movies/2"
  >
    <button
      id={2}
    >
      Explore Characters
    </button>
  </Link>
</div>
`;
